project/plugins/slack/__init__.py

```diff
@@ -8,34 +8,50 @@ def inviteUser(email,configMap,allPermissions, plugin_tag, name):
 
     log = 'Slack: Instruction sent in email.\n'
     instruction = inviteMessage(configMap,plugin_tag)
-    return getJsonResponse('Slack', email, log, instruction)
+    done = True
+    return getJsonResponse('Slack', email, log, instruction, done)
 
 def removeUser(email,configMap,allPermissions, plugin_tag):
+    instruction = ''
+    log = ''
+    done = False
+    userName = email.split('@', 1)[0]
     #get team id
     team = requests.get("https://slack.com/api/team.info?token=" + getApiToken(configMap,plugin_tag) )
     my_json = team.content.decode('utf8')
     data = json.loads(my_json)
-    if not 'error' in data and 'team' in data:
+
+    if 'error' in data:
+        if 'invalid_auth' in team.text:
+            log = plugin_tag + ' error: for user: ' + userName + '. Wrong api token'
+
+        if 'not_authed' in team.text:
+            log = plugin_tag + ' error: for user: ' + userName + '. No api token provided'
+
+        if 'fatal_error' in team.text:
+            log = plugin_tag + ' unexpected error: for user: ' + userName + '. could not be removed'
+
+    else:
         teamId=data['team']['id']
```

Comment:
We should probably still check that we have a team field before trying to dereference it (looks like Eric was checking it above)